### PR TITLE
mpi/coll: Log pipeline information after calculating

### PR DESCRIPTION
**Pull request by raffenet**
## Pull Request Description

Consolidate pipeline information logging into a single location. Fix associated -Wformat warnings. Alternative to pmodels/mpich#5940.

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by hzhou** [comment]
Reference issue #5937

File: src/mpi/coll/algorithms/common/algo_common.h [hunk 1]
```diff
@@ -51,6 +51,15 @@ static inline int MPIR_Algo_calculate_pipeline_chunk_info(MPI_Aint chunk_size, M
     *num_segments = (count + *segsize_ceil - 1) / (*segsize_ceil);
 
   fn_exit:
+    MPL_DBG_MSG_FMT(MPIR_DBG_COLL, VERBOSE,
+                    (MPL_DBG_FDEST,
```
Comment (sagarth) [inline src/mpi/coll/algorithms/common/algo_common.h:55]: We could possibly pass and preserve the calling function name as well. (e.g., "AllReduce", "Reduce" etc.). However, I'm not sure how important that is. Also it seems the function entry /exit log can add that info. Any thoughts?

Comment (raffenet) [reply src/mpi/coll/algorithms/common/algo_common.h:55]: > ...it seems the function entry /exit log can add that info. Any thoughts?

This was my thinking for not including it. Just look at the previous log messages for context. Plus it doesn't add an argument argument to the pipeline calculation API that's unused by anything but the log output.

Comment (sagarth) [reply src/mpi/coll/algorithms/common/algo_common.h:55]: That sounds fine.

**Review by sagarth — APPROVED** [review]

**Pull request merged**